Framework/Algorithms/src/SortXAxis.cpp

```diff
@@ -70,7 +70,7 @@ void SortXAxis::exec() {
   const bool isAProperHistogram =
       (!ignoreHistogramValidation)
           ? determineIfHistogramIsValid(*inputWorkspace)
-          : false;
```

Comment:
I think this expression deserves an inline comment. Also you should add a unit test for this, since it was apparently not captured.

Reply:
It was only ever captured in Debug builds, so a automated unit test wouldn't show this. If that is what you mean.

Reply:
Thanks for adding the comment. I have hard time understanding how can this code change have effect only on debug builds. Can you elaborate more?

Reply:
I have not come across this on anything but Windows, Debug. Maybe @robertapplin can go into the detail as I do not remember the detail from his personal build investigation.

Reply:
I think the logic expression can be simplified:
```
 const bool isAProperHistogram = ignoreHistogramValidation || determineIfHistogramIsValid(*inputWorkspace);
```

Reply:
I have only been able to replicate this on windows, debug mode. There was a vector subscript out-of-range error as the `workspaceIndices` vector was not having its largest value removed for a histogram workspace when `IgnoreHistogramValidation` was set to true. The reason why the same error doesn't occur on release mode I do not know.